### DISCUSSION TITLE
Search collection items by metadata (datasource path)

### DISCUSSION THREAD

**Discussion by irfansuteja** [tryAGI/LangChain#441] [Q&A]
Hi,

In the latest version, is there a function that allows searching a vector collection based on metadata? When using the function vectorDatabase.AddDocumentsFromAsync<PdfPigPdfLoader>(), the source path is stored in the metadata.

If a file is split into multiple documents, say five, how can I update the related files? Specifically, I need to delete the vector documents that share the same metadata/path. Is there a function for this, or is there a better approach?

Thanks.

**Comment by robalexclark** [comment]
I sort of have a similar question. There doesn't seem to be a way to only add new documents to a vector store, its either delete everything and start again (AddDocumentsToDatabaseBehavior.OverwriteExistingCollection) or add the documents again even if they are always added (AddDocumentsToDatabaseBehavior.AlwaysAddDocuments).

I think we need a new option AddDocumentsToDatabaseBehavior.AddNewDocumentsOnly, that only adds documents if they don't already exist.

I've got a feeling though that the path might need its own table column to retrieve if efficiently to do the check if it already exists.

I'm happy to attempt a PR on this if that is something that would be useful.

**Reply by HavenDV** [reply]
Yes, PR on this would be very helpful, thanks in advance!
